src/pages/home/ReportScreen.js

```diff
@@ -348,6 +349,12 @@ function ReportScreen({
         // Report ID will be empty when the reports collection is empty.
         // This could happen when we are loading the collection for the first time after logging in.
         if (!ReportUtils.isValidReportIDFromPath(reportIDFromPath)) {
+            const parentReport = ReportUtils.getReport(report);
+
+            if (ReportUtils.isThread(report) && isEmptyObject(parentReport)) {
+                Report.openReport(report.parentReportID);
+            }
+
```

Comment:
This still does not look right. Why are we opening the parent report if the report id from path is not valid?

Reply:
```
if (report.reportID && report.reportID === getReportID(route) && !isLoadingInitialReportActions) {
    const parentReport = ReportUtils.getReport(report)

    if (ReportUtils.isThread(report) && isEmptyObject(parentReport)) {
        Report.openReport(report.parentReportID)
    }
    
    return;
}
```

Reply:
@s77rt  Do you approve of the above options?